hatch/signing.py

```diff
@@ -92,15 +95,25 @@ def check_expiry(cls, expiry):
             raise ValueError(f"token expired on {expiry.isoformat()}")
         return expiry
 
+    @root_validator
+    def validate_all(cls, values):
+        """If only invalid field is 'expiry',  raise a different exception."""
+        # values dict contains all fields that have been validated.
+        if "expiry" in values:
+            return values
+        if all(k in values for k in ["url", "user", "scope"]):
+            raise cls.Expired()
+        return values
+
     def sign(self, signer=None):
-        if signer is None:
+        if signer is None:  # pragma: no cover
```

Comment:
Providing alternate signers is no longer needed for testing, since I added 'create_raw_token`. But I've left them in as I think it will be useful in future, but no covered for now